### PR TITLE
Implement TextInput.clearTextOnFocus for Android

### PR DESCRIPTION
**Pull request by dgladkov**
Does not fire onSelectionChanged when gaining focus, same as on iOS

### CODE REVIEW

File: Libraries/Components/TextInput/TextInput.js [hunk 1]
```diff
@@ -284,7 +284,6 @@ var TextInput = React.createClass({
     clearTextOnFocus: PropTypes.bool,
     /**
      * If true, all text will automatically be selected on focus
-     * @platform ios
```
Comment (dmmiller) [inline Libraries/Components/TextInput/TextInput.js]: I think you mean to remove this line from the comments above.  You removed it from the `selectTextOnFocus` property

**Comment by dmmiller** [comment]
Can you say a little more about how you tested?  Does js get the callback that the text has changed to ""?   Just because iOS doesn't send selectionChange on focus change when clearing doesn't mean we shouldn't on Android.  I might consider that more a bug and Android doing the right thing.

File: ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java [hunk 2]
```diff
@@ -172,8 +174,12 @@ protected void onSelectionChanged(int selStart, int selEnd) {
   protected void onFocusChanged(
       boolean focused, int direction, Rect previouslyFocusedRect) {
     super.onFocusChanged(focused, direction, previouslyFocusedRect);
-    if (focused && mSelectionWatcher != null) {
-      mSelectionWatcher.onSelectionChanged(getSelectionStart(), getSelectionEnd());
+    if (focused) {
+      if (mClearTextOnFocus) {
+        setText("");
```
Comment (dmmiller) [inline ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java]: Is this the right call?  This certainly will update the EditText, but will it properly update js?

**Comment by dmmiller** [comment]
I'm heading on holiday for 2 weeks, but @andreicoman11 will be a good reviewer for this.

**Comment by dgladkov** [comment]
@mmiller Thanks for your comments! I tested it on the sample project with a TextInput that has `onSelectionChange` event handler with `console.log` call.

JS callback gets called when text is changed to "" and on every focus event except when clearTextOnFocus is set on iOS, so I agree that Android does the right thing in this case. I did this condition check for consistent behavior across platforms, but dispatching selection change event on iOS indeed seems like a better solution.

I'll try to investigate iOS behavior later today.

File: ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java (hunk 2 above)
Comment (dgladkov) [inline ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java]: I did not test if onChange is fired and forgot to test controlled TextInputs behavior, will do that. Thanks for the suggestion!

Comment (andreicoman11) [inline ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java]: `setText()` is the devil and should not be used. You can use `getText().replace(...);` similar to what `maybeSetText()` is doing.

**Comment by mkonicek** [comment]
c

**Comment by facebook-github-bot-4** [comment]
@dgladkov do you have any updates for this pull request? It's been a while so we wanted to check in and see if you've looked at the requested changes.

**Comment by lacker** [comment]
It looks like this PR has been abandoned so I am going to close it. It seems like a useful fix though so please feel free to reopen if you would like to continue working on this!

**Pull request closed without merging**